pkg/app/piped/executor/kubernetes/kubernetes.go

```diff
@@ -605,3 +606,130 @@ func annotateConfigHashToDeployment(m provider.Manifest, managedConfigMaps, mana
 	)
 	return nil
 }
+
+type patcher func(m provider.Manifest, cfg config.K8sResourcePatch) (*provider.Manifest, error)
+
+func patchManifests(manifests []provider.Manifest, patches []config.K8sResourcePatch, patcher patcher) ([]provider.Manifest, error) {
+	if len(patches) == 0 {
+		return manifests, nil
+	}
+
+	out := make([]provider.Manifest, len(manifests))
+	copy(out, manifests)
+
+	for _, p := range patches {
+		target := -1
+		for i, m := range out {
+			if m.Key.Kind != p.Target.Kind {
```

Comment:
Also, isn't it needed to have a check for `APIVersion`?

Reply:
Yeah. We are using the shared `K8sResourceReference` struct around and since it is not including `APIVersion` so that check will not be implemented this time. But that check at all places can be added later when having requests from users.

Reply:
Got it. When it needs that, everything needs to be changed.